js/views/userPage/TypeFilter.js

```diff
@@ -0,0 +1,91 @@
+/* categoryFilter.js and typeFilter.js share quite a bit. Ensure that when one is updated,
+the other is also maintained.
+*/
+
+import $ from 'jquery';
+import _ from 'underscore';
+import loadTemplate from '../../utils/loadTemplate';
+import BaseVw from '../baseVw';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    const opts = {
+      ...options,
+    };
+
+    opts.initialState = {
+      types: [],
+      selected: 'all',
+      expanded: false,
+      maxInitiallyVisibleTypes: 6,
+      ...(options.initialState || {}),
+    };
+
+    super(opts);
+    this.options = opts;
+    this._state = opts.initialState;
+  }
+
+  className() {
+    return 'clrP clrBr padMd clrT contentBox clrSh2 form veryCompact categoryOrTypeFilter';
+  }
+
+  events() {
+    return {
+      'click .js-showMoreLess': 'onClickShowMoreLess',
+      'change input[type="radio"]': 'onChangeType',
+    };
+  }
+
+  onClickShowMoreLess() {
+    this.setState({ expanded: !this.getState().expanded });
+  }
+
+  onChangeType(e) {
+    this._state.selected = e.target.value;
+    this.trigger('type-change', { value: $(e.target).val() });
+  }
+
+  get selectedType() {
+    return this._selected;
+  }
+
+  getState() {
```

Comment:
@jashot7 I'll give this a full review next week, but one quick mention is that setState, getState and replaceState don't need to be declared here. They are already available via the baseVw which all views inherit from. The one caveat is that there is no explicit replaceState, but you can all setState with a `replace` option, e.g. `setState(newState, { replace: true })`.

(I know there are a lot of views in the codebase that implement these methods. That's because for the longest time we were duplicating the code everywhere. When we finally put them in the baseVw, we didn't remove them from everywhere because we weren't sure if any had custom logic. So... we took the stance that going forward we'd use the baseVw ones (unless we needed to override with custom functionality) and for old ones we'd refactor as we went along.)